conjure-java-jackson-serialization/src/test/java/com/palantir/conjure/java/serialization/ObjectMappersTest.java

```diff
@@ -136,6 +140,91 @@ public void testMapWithDuplicateKeys() {
                         + "Key 'test' values ['bar', 'foo']");
     }
 
+    @Test
+    public void testLongDeserializationFromString() throws IOException {
+        assertThat(MAPPER.readValue("\"1\"", Long.class)).isEqualTo(1L);
+    }
+
+    @Test
+    public void testLongTypeDeserializationFromString() throws IOException {
+        assertThat(MAPPER.readValue("\"1\"", Long.TYPE)).isEqualTo(1L);
+    }
+
+    @Test
+    public void testOptionalLongTypeDeserializationFromString() throws IOException {
+        assertThat(MAPPER.readValue("\"1\"", OptionalLong.class)).hasValue(1L);
+    }
+
+    @Test
+    public void testLongDeserializationFromJsonNumber() throws IOException {
+        assertThat(MAPPER.readValue("1", Long.class)).isEqualTo(1L);
+    }
+
+    @Test
+    public void testOptionalLongDeserializationFromJsonNumber() throws IOException {
+        assertThat(MAPPER.readValue("1", OptionalLong.class)).hasValue(1L);
+    }
+
+    @Test
+    public void testLongTypeDeserializationFromJsonNumber() throws IOException {
+        assertThat(MAPPER.readValue("1", Long.TYPE)).isEqualTo(1L);
+    }
+
+    @Test
+    public void testLongDeserializationFromJsonNull() throws IOException {
+        assertThat(MAPPER.readValue("null", Long.class)).isNull();
+    }
+
+    @Test
+    public void testOptionalLongDeserializationFromJsonNull() throws IOException {
+        assertThat(MAPPER.readValue("null", OptionalLong.class)).isEmpty();
+    }
+
+    @Test
+    public void testLongOverflowDeserialization() {
+        BigInteger large = BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE);
+        assertThatThrownBy(() -> MAPPER.readValue("" + large, Long.TYPE))
+                .isInstanceOf(JsonParseException.class)
+                .hasMessageContaining("out of range of long");
+    }
+
+    @Test
+    public void testLongAsStringOverflowDeserialization() {
+        BigInteger large = BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE);
+        assertThatThrownBy(() -> MAPPER.readValue("\"" + large + "\"", Long.TYPE))
+                .isInstanceOf(NumberFormatException.class);
```

Comment:
Should this be a JsonParseException constricted with deserialization context?

Reply:
That's a good question. Existing validation for primitive values does produce JsonParseException, however other input value validation (OffsetDateTime, ResourceIdentifier, etc) does not produce JsonParseException.